### ISSUE TITLE
Question : FPS rate 

### ISSUE THREAD

**Issue by dhgokul** [infocom-tpo/SwiftOpenPose#10]
@mganeko How much time it would take to process live video frames @ 60 FPS ?

**Comment by mbotsu** [comment]
Hi.
If it is a high-speed device it can process in about 0.2 seconds.
That is, it can be processed with 12 sec = 0.2 sec * 60 frame.
Thanks.

**Comment by dhgokul** [comment]
Thanks mbotsu !

**Comment by dhgokul** [comment]
@mganeko is there any way to speed up detection process ?

**Comment by chen-xi-wang** [comment]
Yes, same question as @dhgokul. I am considering that whether this project could use Metal performance shaders to do the part of Matrix Multiplication to improve the process speed.   if you guys have any way to speed up it, please to let me know. Thanks @mganeko and @dhgokul !

**Comment by mbotsu** [comment]
Hi.

Much of the cost is the structure of the machine learning model and the problem of the PAF algorithm.
There is PoseNet as a new approach.
However, this also has a problem of accuracy.
Please try [nfocom-tpo / PoseNet-CoreML](https://github.com/infocom-tpo/PoseNet-CoreML) if you like.
Thanks.

**Comment by chen-xi-wang** [comment]
Ok. Thanks a lot!

**Comment by chen-xi-wang** [comment]
Hi mbotsu, I'm still working on this project, that is awesome. but some points of the code I really confused. Could you please provide a development document or some explanation like flow chart? That would be very helpful! Thanks!

**Comment by NarutoYq** [comment]
Estimator.swift's estimate method uses a Matrix structure, which is a waste of time. Especially during initialization, the matrix data will be copied internally, which can be replaced with an Array and accessed through a pointer. After the last optimization, the same picture needs 200ms before optimization, and only 50ms after optimization。

**Comment by mbotsu** [comment]
Hi.

I haven't tried it yet, but if you want a faster implementation I recommend this code.
Apache License 2.0

https://github.com/opencv/open_model_zoo/tree/master/demos/human_pose_estimation_demo

and CoreML Model Convert Log.
https://gist.github.com/mbotsu/8c712093884cb64f9477881cb182dbc2

**Comment by mbotsu** [comment]
CoreML Model Convert Log
https://gist.github.com/mbotsu/8c712093884cb64f9477881cb182dbc2

**Comment by NarutoYq** [comment]
Thank you
